### ISSUE TITLE
CMake is not currently configuring correct

### ISSUE THREAD

**Issue by crtrott** [kokkos/kokkos#2343]
Assuming the GNU Makefiles are the truth, then CMake doesn't do the right thing. To test this run the scripts associated with pull request #2340 and issue #2333 
We should fix this first, and then add all the other bells and whistles.

**Comment by dalg24** [comment]
PRs to address this current issue should fetch `crtrott:issue-2333` (that has the changes from #2340) and rebase it on develop.
```
git remote add upstream https://github.com/kokkos/kokkos.git
git remote add crtrott https://github.com/crtrott/kokkos.git
git fetch crtrott issue-2333
git checkout issue-2333
git rebase upstream/develop
```

Then go to the branch with your changes and rebase it on top of `issue-2333` and run the script
```
git checkout <your_branch_name_here>
git rebase issue-2333
<your_kokkos_path_here>/core/unit_test/configuration/test-code/test_config.bash <your_kokkos_path_here>
```
and make sure it does not print anything.  When you are happy with your changes rebase on top `develop` and push to your fork to open a PR.

```
git rebase upstream/develop
git push <your_fork> <your_branch_name_here>
```

**Comment by masterleinad** [comment]
For anyone looking into this, it makes sense to disable
```
+CHECK_DEVICE_SPECIFIC_OPTIONS(DEVICE CUDA OPTIONS CUDA_UVM CUDA_RELOCATABLE_DEVICE_CODE CUDA_LAMBDA CUDA_LDG_INTRINSIC)
+CHECK_DEVICE_SPECIFIC_OPTIONS(DEVICE HPX OPTIONS HPX_ASYNC_DISPATCH)
```
in `cmake/kokkos_enable_options.cmake`. Such that the configuration succeeds even if it is invalid.

**Comment by jjwilke** [comment]
Yeah, I'm going to add an option `Kokkos_SKIP_VALIDATION" that will get passed to the tests.

**Comment by jjwilke** [comment]
I see the main issue. All of the arch if statements from the Makefile did get transferred, but only for setting the flag - not the feature macros.

**Comment by PhilMiller** [comment]
@crtrott The CMake build is now treated as authoritative. The commits referencing this are long-since merged, in 3.0. Could you confirm that this was addressed and close it?

**Comment by cwpearson** [comment]
Irrelevant since #8374
